### PR TITLE
Dat 3139 fix infobox params API

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/DAT-3139

### CODE REVIEW

File: extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php [hunk 1]
```diff
@@ -113,13 +113,32 @@ protected function getInfoboxes( $text ) {
...
 	 * @param $text string template text
-	 * @return mixed
+	 * @return string
 	 */
 	protected function getIncludeonlyText( $text ) {
 		preg_match_all( "/<includeonly>(.+)<\/includeonly>/sU", $text, $result );
 
-		return isset( $result[1][0] ) ? $result[1][0] : false;
+		if ( !isset( $result[1][0] ) ) {
+			return null;
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php:125]: This is not consistent with previous approach, why the change?

File: extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php [hunk 2]
```diff
@@ -113,13 +113,32 @@ protected function getInfoboxes( $text ) {
...
+		return $result;
+
+	}
+
+	/**
+	 * @desc for given template text returns it without text in <nowiki> and <pre> tags
+	 * @param $text string
+	 * @return string
+	 */
+	protected function removeNowikiPre( $text ) {
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php:138]: Are you sure those are the only cases?

File: extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php (hunk 1 above)
Comment (dianafa) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php:125]: we don't want to proceed if there are no includeonly tags

Comment (idradm) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php:125]: Previously you were returning false here, right now it's null. Is there a reason for this change? Will the code behave the same way?

**Comment by idradm** [comment]
:ship: it

**Comment by dianafa** [comment]
tested, merging.

Failing rss test not related.
